M2/Macaulay2/packages/Macaulay2Doc/functions/replace-doc.m2

```diff
@@ -21,29 +21,40 @@ doc ///
     re:String
       a @TO2 {"regular expressions", "regular expression"}@ describing a pattern
     replacement:String
-      a replacement or regex formatting string, may include backreferences
+      following the @TO2 {"regular expressions", "formatting syntax"}@
     str:String
       a subject string to be processed
     POSIX=>Boolean
       if true, interpret the @TT "re"@ using the POSIX Extended flavor, otherwise the Perl flavor
   Outputs
     :String
-      the string obtained from @TT "str"@ by replacing its substrings matching @TT "re"@ by copies of @TT "replacement"@.
+      obtained from @TT "str"@ by replacing its substrings matching @TT "re"@ by copies of @TT "replacement"@
```

Comment:
Same as the others. This is correct. In the output, it will say "a string, obtained from ...". Declaring the type as String and then immediately saying that it is "a string" is redundant.

Reply:
Maybe we should get rid of that pesky comma, which is automatically inserted.

Reply:
I would be in favor of a different type of output, like:
> X : String -- blah blah

But for now, this seems fine:
![image](https://user-images.githubusercontent.com/3147718/92021465-6dcad580-ed1f-11ea-8144-e8081f9e4c30.png)

I think a much needed improvement is more fine grained type checking and type printing, for instance printing "a list of lists of integers" or "a list of ideals" instead of just "a list".

Reply:
> I think a much needed improvement is more fine grained type checking and type printing, for instance printing "a list of lists of integers" or "a list of ideals" instead of just "a list".

That information, however, is not available.